sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -2102,6 +2102,18 @@ val SHUFFLE_COMPRESSION_LZ4_CHUNK_SIZE = conf("spark.rapids.shuffle.compression.
       .booleanConf
       .createWithDefault(false)
 
+  val SHUFFLE_COALESCE_BEFORE_SHUFFLE_TARGET_SIZE_RATIO =
+    conf("spark.rapids.shuffle.coalesceBeforeShuffleTargetSizeRatio")
+      .doc("Ratio of target batch size for coalescing batches before shuffle partitioning. " +
+        "The target size for coalescing is calculated as: " +
+        "targetBatchSize * coalesceBeforeShuffleTargetSizeRatio. " +
+        "A value of 0.6 means the coalesce target is 60% of the configured batch size. " +
+        "A value of 0 disables coalescing before shuffle.")
+      .internal()
+      .doubleConf
+      .checkValue(v => v >= 0.0 && v <= 1.0, "Ratio must be between 0.0 and 1.0")
```

Comment:
can we get an idea of what happens when you go from 0 to 1 in the coalesce? Why would we not always set it to 1?

Reply:
Hi @abellina so simplify the logic I removed this config, and choose to always insert a coalesce before shuffle